pyro/distributions/transforms/reshape.py

```diff
@@ -0,0 +1,84 @@
+import torch
+from torch.distributions import constraints
+from torch.distributions.transforms import Transform
+
+from pyro.distributions.util import copy_docs_from
+
+
+@copy_docs_from(Transform)
+class Reshape(Transform):
+    """
+    A bijection that reshapes the event of the input into an arbitrary-size Tensor. The primary intended use of this
+    transform is to allow interoperatability between transforms on 1D vector random variables and 2D image random
+    variables.
+
+    Note that this transform may convert batch dimensions into event dimensions, depending on the input/output shapes.
+    The event dim of the output of this transformation will be the length of the `to_event_shape` argument.
+
+    Example usage:
+
+    >>> c = 3 # assume 3 x 32 x 32 RGB inputs
+    >>> base_dist = dist.Normal(torch.zeros(c*32*32), torch.ones(c*32*32))
```

Comment:
I suppose you'll have to call `.to_event()` so that the three dims are considered event dims and not batch dims.